Libplanet.Tests/Net/Messages/NetMQMessageCodecTest.cs

```diff
@@ -0,0 +1,134 @@
+using Libplanet.Blockchain.Policies;
+using Libplanet.Blocks;
+using Libplanet.Crypto;
+using Libplanet.Net;
+using Libplanet.Net.Messages;
+using Libplanet.Store;
+using Libplanet.Tests.Common.Action;
+using Libplanet.Tests.Store.Trie;
+using NetMQ;
+using Xunit;
+
+namespace Libplanet.Tests.Net.Messages
+{
+    public class NetMQMessageCodecTest
+    {
+        [Fact]
+        public void Encode()
+        {
+            var privateKey = new PrivateKey();
+            var peer = new Peer(privateKey.PublicKey);
+            var dateTimeOffset = DateTimeOffset.UtcNow;
+            var appProtocolVersion = new AppProtocolVersion(
+                1,
+                new Bencodex.Types.Integer(0),
+                ImmutableArray<byte>.Empty,
+                default(Address));
+            var message = new Ping();
+            var codec = new NetMQMessageCodec();
+            NetMQMessage raw =
+                codec.Encode(message, privateKey, peer, dateTimeOffset, appProtocolVersion);
+            var parsed = codec.Decode(raw, true, (i, p, v) => { }, null);
+            Assert.Equal(peer, parsed.Remote);
+            Assert.Equal(appProtocolVersion, parsed.Version);
+            Assert.Equal(dateTimeOffset, parsed.Timestamp);
+        }
+
+        [Theory]
+        [InlineData(Message.MessageType.Ping)]
+        [InlineData(Message.MessageType.Pong)]
+        [InlineData(Message.MessageType.GetBlockHashes)]
+        [InlineData(Message.MessageType.TxIds)]
+        [InlineData(Message.MessageType.GetBlocks)]
+        [InlineData(Message.MessageType.GetTxs)]
+        [InlineData(Message.MessageType.Blocks)]
+        [InlineData(Message.MessageType.Tx)]
+        [InlineData(Message.MessageType.FindNeighbors)]
+        [InlineData(Message.MessageType.Neighbors)]
+        [InlineData(Message.MessageType.BlockHeaderMessage)]
+        [InlineData(Message.MessageType.BlockHashes)]
+        [InlineData(Message.MessageType.GetChainStatus)]
+        [InlineData(Message.MessageType.ChainStatus)]
+        [InlineData(Message.MessageType.DifferentVersion)]
+        public void CheckMessages(Message.MessageType type)
+        {
+            var privateKey = new PrivateKey();
+            var peer = new Peer(privateKey.PublicKey);
+            var dateTimeOffset = DateTimeOffset.UtcNow;
+            var appProtocolVersion = new AppProtocolVersion(
+                1,
+                new Bencodex.Types.Integer(0),
+                ImmutableArray<byte>.Empty,
+                default(Address));
+            var message = CreateMessage(type);
+            var codec = new NetMQMessageCodec();
+            NetMQMessage raw =
+                codec.Encode(message, privateKey, peer, dateTimeOffset, appProtocolVersion);
+            var parsed = codec.Decode(raw, true, (i, p, v) => { }, null);
+            Assert.Equal(appProtocolVersion, parsed.Version);
+            Assert.Equal(peer, parsed.Remote);
+            Assert.Equal(dateTimeOffset, parsed.Timestamp);
```

Comment:
Maybe just adding
```csharp
Assert.Equal(message.DataFrames, parsed.DataFrames);
```
will do. Do you think we message should implement `IEquatable`? IMO it will be only useful during tests.

Reply:
> Do you think we message should implement `IEquatable`? IMO it will be only useful during tests.

I agree.  For the moment, it's enough with comparing data frames IMHO.